### PR TITLE
Refine examples via unifying device settings and output folder names

### PR DESCRIPTION
**Pull request by xianyuanliu**
### Description
Refine examples via unifying parameter names, device settings, and output directory names. These changes do not change any structure of the model. Theoretically, they do not affect the model. If there is any effect, please report via raising an issue.

1. Set the output directory parameter as cfg.OUTPUT.OUT_DIR.
2. Set the output directory as "outputs".
3. Use the parameters 'accelerator' and 'devices' to control the device settings, following the PyTorch 2.0+ update.
4. Make the config file style more consistent, by adding comments and unifying some parameter names (CN -> CfgNode).

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

File: examples/action_dann/main.py [hunk 1]
```diff
@@ -93,12 +92,10 @@ def main():
...
-            devices=args.devices,
-            logger=tb_logger,  # logger,
-            # weights_summary='full',
-            fast_dev_run=cfg.OUTPUT.FAST_DEV_RUN,  # True,
+            devices=args.devices if args.devices != 0 else "auto",
+            logger=tb_logger,
+            fast_dev_run=cfg.OUTPUT.FAST_DEV_RUN,
             callbacks=[lr_monitor, checkpoint_callback, progress_bar],
-            # callbacks=[early_stop_callback, lr_monitor],
             # limit_train_batches=0.005,
```
Comment (shuo-zhou) [inline examples/action_dann/main.py:99]: Is it necessary to keep these comments?

File: examples/avmnist_multimodal/main.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def arg_parse():
     parser.add_argument(
         "--gpus",
         default=0,
-        help="gpu id(s) to use. None/int(0) for cpu. list[x,y] for xth, yth GPU."
+        help="gpu id(s) to use. int(0) for cpu. list[x,y] for xth, yth GPU."
         "str(x) for the first x GPUs. str(-1)/int(-1) for all available GPUs",
```
Comment (shuo-zhou) [inline examples/avmnist_multimodal/main.py:32]: Why does `dafault=0` here and `dafault=1` in other examples, explain `default=0`?

File: examples/polypharmacy_gripnet/config.py [hunk 3]
```diff
@@ -1,55 +1,57 @@
...
-C.GRIPN_SV1.NAME = "protein"
-C.GRIPN_SV1.INTER_FEAT_CHANNELS = 16
-C.GRIPN_SV1.INTER_AGG_CHANNELS_LIST = [16, 16]
-C.GRIPN_SV1.EXTER_AGG_CHANNELS_LIST = []
+_C.GRIPN_SV1 = CfgNode()
+_C.GRIPN_SV1.NAME = "protein"
+_C.GRIPN_SV1.INTER_FEAT_CHANNELS = 16
+_C.GRIPN_SV1.INTER_AGG_CHANNELS_LIST = [16, 16]
+_C.GRIPN_SV1.EXTER_AGG_CHANNELS_LIST = []
 # Elements in `EXTER_AGG_CHANNELS_LIST` should be a list of combinations of supervertex name (str) and embedding
```
Comment (shuo-zhou) [inline examples/polypharmacy_gripnet/config.py:38]: @peizhenbai Are these commented lines needed?

**Review by shuo-zhou — CHANGES_REQUESTED** [review]
Looks good to me, with a few minor changes required.

File: examples/avmnist_multimodal/main.py (hunk 2 above)
Comment (xianyuanliu) [reply examples/avmnist_multimodal/main.py:32]: It is set to 0 originally here, i.e. using CPU only. I did not change it due to my unfamiliarity with this example.

File: examples/action_dann/main.py (hunk 1 above)
Comment (xianyuanliu) [reply examples/action_dann/main.py]: Removed

File: examples/polypharmacy_gripnet/config.py (hunk 3 above)
Comment (peizhenbai) [reply examples/polypharmacy_gripnet/config.py:38]: Yes, the comments are left to explain the function of EXTER_AGG_CHANNELS_LIST. I think we can keep it if there are no other concerns.

**Review by shuo-zhou — APPROVED** [review]

**Pull request merged**